b2/src/main/java/org/jclouds/b2/B2ApiMetadata.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.labs.b2;
+
+import static org.jclouds.reflect.Reflection2.typeToken;
+import java.net.URI;
+import java.util.Properties;
+
+import org.jclouds.apis.ApiMetadata;
+import org.jclouds.blobstore.BlobStoreContext;
+import org.jclouds.labs.b2.config.B2HttpApiModule;
+import org.jclouds.rest.internal.BaseHttpApiMetadata;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Module;
+
+@AutoService(ApiMetadata.class)
+// TODO: B2ProviderMetadata?
```

Comment:
Yes. The api metadata should just describe the API, and the provider metadata should add to the API the real endpoint (apis can provide a default one), and the default configuration. For providers that have their own API this makes less sense, but I think it is still good to separate both. HAve a look at the digitalocean2 provider for an example.

You may also want to use the same provider id for the api and provider, so just remove this annotation once the provider metadata exists, to avoid having conflicts.